test/Microsoft.AspNetCore.Server.Kestrel.Performance/RequestParsing.cs

```diff
@@ -122,34 +108,37 @@ public void ParseUnicodePipelined()
         {
             for (var i = 0; i < InnerLoopCount; i++)
             {
-                InsertData(_unicodePipelinedRequests);
-
-                ParseData();
+                ParseData(_unicodePipelinedRequests);
             }
         }
 
-        private void InsertData(byte[] dataBytes)
+        private void ParseData(byte[] dataBytes)
         {
-            SocketInput.IncomingData(dataBytes, 0, dataBytes.Length);
-        }
+            // We need to copy because we modify data inplace in TakeStartLine
+            var data = new byte[dataBytes.Length];
+            Array.Copy(dataBytes, data, data.Length);
 
-        private void ParseData()
-        {
-            while (SocketInput.GetAwaiter().IsCompleted)
+            var readableBuffer = ReadableBuffer.Create(data);
```

Comment:
I would like to continue measuring the cost of SocketInput/Pipelines in these benchmarks.